### DISCUSSION TITLE
findByCursor count inaccurate, count 50 but only return 48

### DISCUSSION THREAD

**Discussion by duyvinh229148** [mikro-orm/mikro-orm#5988] [General]
### Describe the bug

This is my function
```
  async searchHoaDonTong(ownerId: number, payload: SearchHoaDonTongDto) {
    const {
      tkHddtId,
      fromDate,
      toDate,
      sellBuyType,
      endCursor,
      sortDescriptor,
    } = payload;

    const currentCursor = await this.hoaDonRepository.findByCursor(
      {
        taiKhoanHddt: tkHddtId,
        sellBuyType,
        tdlap: {
          $gte: getDbFilterHoaDonDate(fromDate),
          $lte: getDbFilterHoaDonDate(toDate),
        },
      },
      {
        first: DbConfig.limit,
        after: endCursor,
        orderBy: this.getSorter(sortDescriptor),
        fields: [
          'id',
          'shdon',
          'khhdon',
          'khmshdon',
          'tdlap',
          'nbten',
          'nbmst',
          'nmmst',
          // 'nbsdthoai',
          'nmten',
          'tgtcthue',
          'tgtthue',
          'tgtttbso',
          'trangthaihd',
          'nbcks',
        ],
        populate: [
          'hoaDonFiles.fileType',
          'hoaDonFiles.status',
          'hoaDonFiles.linkTraCuu',
          'hoaDonFiles.maTraCuu',
          'hoaDonFiles.nhaCungCap',
          'taiKhoanHddt.id',
        ],
      },
    );

    return {
      items: currentCursor.items,
      hasNextPage: currentCursor.hasNextPage,
      endCursor: currentCursor.endCursor,
      totalCount: currentCursor.totalCount,
    };
  }

```
I use infinitive query, each request will return 20 records
![image](https://github.com/user-attachments/assets/7ccb559b-d7fb-45a2-9f29-f61899c1bdb3)
![image](https://github.com/user-attachments/assets/4e23c621-f2f5-4192-8d03-a0f727a9d48d)
![image](https://github.com/user-attachments/assets/b67c9f0c-24c2-4626-87f1-0bda13df49be)

Only 48 records is returned, although findByCursor say 50

### Reproduction

N/A

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

^6.3.4

### Node.js version

18

### Operating system

Linux - Fedora 39

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> This is my function

That is not a reproduction, I need to see code I can run. Also hard to help with screenshots, you don't have to convince me there is a problem, you need to show the problem to me so I can fix it, and you can't do that with screenshots. Please provide a [complete but minimal reproduction](https://github.com/mikro-orm/reproduction). 

> ^6.3.4

That is not a version, that is a version range. If you are not using latest, upgrading should be your first step.

**Comment by duyvinh229148** [comment]
Thank for fast reply. Love your library. Can you suggest me any way that I can work around with this. The database show 50 record, totalCount is 50, but only 48 records is returned. Is there any thing that I can try to overcome this?

**Reply by duyvinh229148** [reply]
I commented out all the 'fields' and 'populate', still same result

**Reply by duyvinh229148** [reply]
I upgraded to latest version 6.3.7 too

**Reply by B4nan** [reply]
Try to create the reproduction, it sounds like a bug, but without that it's hard to tell (and impossible to fix).

Do you use filters? How do the queries look like?

**Reply by duyvinh229148** [reply]
![image](https://github.com/user-attachments/assets/5a768fa8-f4ad-4ad7-a054-6837cb19b931)
here is the query I want, database return 50 rows

**Reply by duyvinh229148** [reply]
![image](https://github.com/user-attachments/assets/321b6ebd-acb3-401c-913e-d3f57e464acd)
findByCursor miss this 2 rows. I suspect that the last cursor is inaccurate, it skip this 2 row in the last cursor find

**Reply by duyvinh229148** [reply]
I can share with you my private repo if that help

**Reply by duyvinh229148** [reply]
I dont use any filter, have never try that feature

**Reply by B4nan** [reply]
so the count is correct, but the items are missing? sounds like your cursor is not unique. the fact that you use `>=` and `<=` there feels quite suspicious, one side of the interval should be open

**Reply by duyvinh229148** [reply]
I can share with you the credential to my prod database too. All for the sake of debugging

**Reply by B4nan** [reply]
i dont want your private repo, i want a minimal reproduction.

**Reply by duyvinh229148** [reply]
perhaps you miss this condition, please watch this:
https://youtu.be/zwDIN04lIpc?si=Y8L2-ugee8N0gVRp&t=633

**Reply by duyvinh229148** [reply]
![image](https://github.com/user-attachments/assets/f86f5abd-8f14-4a12-8f10-11aa8eb4f55e)
my 2 missing items has the same order value

**Reply by duyvinh229148** [reply]
![image](https://github.com/user-attachments/assets/25ec8dc7-3c04-4138-8eb8-d6bae8044517)

**Reply by B4nan** [reply]
let me say it once again, i really need a complete reproduction, not you pointing to the code, or linking a video.
